autonomi/src/uploader/upload.rs

```diff
@@ -75,13 +77,26 @@ pub(super) async fn start_upload(
             // 6 because of the 6 pipelines, 1 for redundancy.
             mpsc_channel(uploader.cfg.batch_size * 6 + 1)
         };
+
     let (make_payment_sender, make_payment_receiver) = mpsc_channel(uploader.cfg.batch_size);
 
-    uploader.start_payment_processing_thread(
-        make_payment_receiver,
-        task_result_sender.clone(),
-        uploader.cfg.payment_batch_size,
-    )?;
+    match &uploader.payment_option {
+        PaymentOption::Wallet(wallet) => {
+            InnerUploader::start_payment_processing_thread(
+                wallet.clone(),
+                make_payment_receiver,
+                task_result_sender.clone(),
+                uploader.cfg.payment_batch_size,
+            )?;
+        }
+        PaymentOption::Receipt(receipt) => {
+            let _ = task_result_sender
+                .send(TaskResult::MakePaymentsOk {
+                    payment_proofs: receipt.clone(),
+                })
+                .await;
```

Comment:
I think we should probably not support `Receipt` insidde the uploader? The uploader would perform get_store cost even if we provide the receipt here, also I'm not really sure how this would work here. It was not designedd with pre-paid data in mind.

I think we should probably make the uploader moduler if we were to support receipts. We should probably just error out here?

Reply:
We need prepaid uploads for the webapp that releases Tuesday, so I had to quick and dirty get support in there. 

Do you have a better idea of how we might do it? Ideally, It must be very quick to do, as I need to get started with coupling this API to the webapp.

Reply:
I took a look at the code again and I think this commit should cover the edge cases.. https://github.com/maidsafe/safe_network/commit/09b0a72dd29a85206812b6c837759389a9a1fd7f

Reply:
Brilliant, thank you!!